### PR TITLE
On tests that specify `--color=always` emit SVG file with stderr output

### PR DESCRIPTION
**Pull request by estebank**
Leverage `anstyle-svg`, as `cargo` does now, to emit `.svg` files instead of `.stderr` files for tests that explicitly enable color output. This will make reviewing changes to the graphical output of tests much more human friendly.

### CODE REVIEW

**Comment by estebank** [comment]
All credit to @epage for the library as well as @Muscraft for showing me https://github.com/rust-lang/cargo/pull/13461.

File: src/tools/compiletest/src/runtest.rs [hunk 1]
```diff
@@ -4014,9 +4014,17 @@ impl<'test> TestCx<'test> {
         explicit_format: bool,
     ) -> usize {
         let stderr_bits = format!("{}bit.stderr", self.config.get_pointer_width());
+        let force_color_svg = self.props.compile_flags.iter().any(|s| s.contains("--color"))
+            && !self.config.target.contains("windows");
```
Comment (compiler-errors) [inline src/tools/compiletest/src/runtest.rs:4018]: Why is this necessary?

Comment (estebank) [reply src/tools/compiletest/src/runtest.rs:4018]: It's not absolutely necessary, but I don't have access to a windows box to ensure this test ran *and* the colors for the ANSI escapes is subtly different between windows and non-windows. Can change that and do a full build on CI to get the windows svg file if we want to land it with it.

Comment (epage) [reply src/tools/compiletest/src/runtest.rs:4018]: For annotate-snippets, @Muscraft  recently added a `testing-colors` feature flag to force the colors the same across all platforms so we don't have to deal with platform-specific colors (except for the places we need to to verify they are platform-specific)

Comment (compiler-errors) [reply src/tools/compiletest/src/runtest.rs:4018]: I'm just worried that when ui testing on windows platforms, this will cause test failures since compiletest expects stdout files but there are svg files here.

Comment (estebank) [reply src/tools/compiletest/src/runtest.rs:4018]: I *believe* that the filter I included was for *explicit* `windows` platform request from the test file. I will remove the platform check.

@epage is `testing-colors` included in 0.1.3?

Comment (epage) [reply src/tools/compiletest/src/runtest.rs:4018]: `testing-colors` is a annotate-snippets feature.  Not really relevant until rustc is converted over but just giving an idea of strategies.

Comment (compiler-errors) [reply src/tools/compiletest/src/runtest.rs:4018]: > I believe that the filter I included was for explicit windows platform request from the test file. I will remove the platform check.

For the record, the variable `self.config.target` is set for all UI tests, so I believe this will produce observable behavioral differences in compiletest, and not just for tests tagged `// only-windows`/`// ignore-windows`.

> I will remove the platform check.

Cool, that sounds like the easiest solution.

**Comment by compiler-errors** [comment]
@rustbot author

should be good to go after fixing and blessing the window test, but i'd like to take a look at the final results

File: src/tools/compiletest/src/runtest.rs (hunk 1 above)
Comment (Muscraft) [reply src/tools/compiletest/src/runtest.rs:4018]: Another solution to this might be to pull the colors out into a separate crate and share them between `rustc` and `annotate-snippets`. I have thought about doing this before, but have never talked with enough people to see if it was a good idea. It would make it easier to share the colors during the transition period and allow other crates to match colors in the long run.

**Comment by compiler-errors** [comment]
For the record, I think `bors try` just builds artifacts for x86_64-unknown-linux-gnu

**Review by compiler-errors — APPROVED** [review]
r=me when ci is green
